### PR TITLE
2020 13 uniref results

### PR DESCRIPTION
**Pull request by xwatkins**
- Made `ResultsContainer` and `ResultsView` components more generic
- Created `ColumnConfiguration` for UniRef
- Re-enabled tests for `ResultsContainer`
- Created more issues under https://www.ebi.ac.uk/panda/jira/browse/TRM-24743 (Action buttons, download functionality)

### CODE REVIEW

**Comment by aurel-l** [comment]
One a first quick look, I think the logic for the "Types" column is flawed somewhere, because all of the entries appear to only have UniProtKB unreviewed members right now.

File: src/uniref/adapters/uniRefConverter.ts [hunk 1]
```diff
@@ -70,7 +64,7 @@ export type UniRefLiteAPIModel = {
   organismCount: number;
   representativeId: string;
   seedId: string;
-  memberIdTypes: MemberIdType;
+  memberIdTypes?: string[];
```
Comment (aurel-l) [inline src/uniref/adapters/uniRefConverter.ts:67]: Not sure why the change to remove the specificity and make it a string. Aren't we supposed to only have 3 possible options here?

File: src/uniref/config/__tests__/__snapshots__/UniRefColumnConfiguration.spec.tsx.snap [hunk 2]
```diff
@@ -0,0 +1,67 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`UniRefColumnConfiguration component should render all columns: Cluster ID 1`] = `
+<DocumentFragment>
+  <a
+    href="/uniref/undefined"
```
Comment (aurel-l) [inline src/uniref/config/__tests__/__snapshots__/UniRefColumnConfiguration.spec.tsx.snap:6]: Something's off here too 🤔

Comment (xwatkins) [reply src/uniref/config/__tests__/__snapshots__/UniRefColumnConfiguration.spec.tsx.snap]: Was passing the wrong object...

File: src/uniref/adapters/uniRefConverter.ts (hunk 1 above)
Comment (xwatkins) [reply src/uniref/adapters/uniRefConverter.ts]: Not sure either. Updated to `EntryType`

**Comment by xwatkins** [comment]
> One a first quick look, I think the logic for the "Types" column is flawed somewhere, because all of the entries appear to only have UniProtKB unreviewed members right now.

I'm able to see some UniProtKB members by searching for P05067 in UniRef. Have merged `2020_13` in the branch and added UniParc icon.

File: src/shared/components/entry/EntryTypeIcon.tsx [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        title={title || 'This marks an unreviewed entry'}
+      >
+        <TremblIcon />
+      </span>
+    );
+  }
+  if (entryTypeToCheck === EntryType.UNIPARC) {
+    return (
+      <span
+        className="entry-title__status icon--unreviewed"
```
Comment (aurel-l) [inline src/shared/components/entry/EntryTypeIcon.tsx:43]: `icon--unreviewed` is wrong here. It either requires a `icon--uniparc` class added in Franklin, or colouring to be applied from the uniprot-website codebase

File: src/shared/components/results/ResultsView.tsx [hunk 4]
```diff
@@ -1,46 +1,88 @@
...
 };
 
-const ResultsView: React.FC<ResultsTableProps> = ({
+const convertRow = (
+  row: UniProtkbAPIModel | UniRefLiteAPIModel,
+  namespace: Namespace
+) => {
+  switch (namespace) {
+    case Namespace.uniprotkb:
+      return uniProtKbConverter(row as UniProtkbAPIModel) as UniProtkbUIModel;
```
Comment (aurel-l) [inline src/shared/components/results/ResultsView.tsx:59]: The 2nd `as` is not needed here

File: src/shared/components/results/ResultsView.tsx [hunk 5]
```diff
@@ -1,46 +1,88 @@
...
+        primaryAccession;
+    case Namespace.uniref:
+      return ({ id }: { id: string }) => id;
+    default:
+      return null;
+  }
+};
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const ColumnConfigurations: Partial<Record<Namespace, Map<any, any>>> = {
```
Comment (aurel-l) [inline src/shared/components/results/ResultsView.tsx:80]: Maybe not needed for now, but maybe add a `TODO:` comment to remove the `any, any` types eventually.
We need a Column type covering UniProtKBColumns, UniRefColumns, etc for the first any, and a Column renderer type with `label: string` and a render definition for the second any.

File: src/shared/components/entry/EntryTypeIcon.tsx (hunk 3 above)
Comment (xwatkins) [reply src/shared/components/entry/EntryTypeIcon.tsx:43]: I'll add an `icon--uniparc`, I think it's more consistent.

File: src/shared/components/results/ResultsView.tsx (hunk 4 above)
Comment (xwatkins) [reply src/shared/components/results/ResultsView.tsx]: Removed

File: src/shared/components/results/ResultsView.tsx (hunk 5 above)
Comment (xwatkins) [reply src/shared/components/results/ResultsView.tsx:81]: Added comment

File: src/shared/components/entry/EntryTypeIcon.tsx (hunk 3 above)
Comment (xwatkins) [reply src/shared/components/entry/EntryTypeIcon.tsx]: Added to franklin, will be available next time we publish

**Review by aurel-l — APPROVED** [review]
